### DISCUSSION TITLE
newbie question for starting a voice assistant

### DISCUSSION THREAD

**Discussion by jevylux** [tayler6000/pyVoIP#223] [Q&A]
Hi, i want to build a voice assistant system on a raspberry pi.
The hardware should be an analogue phone, connected through a Grandstream HT801 to 'something' . The device should record the incomung coice call and send it to a speecht-to-text, probably whisper. I saw the demo code  but it seems as if the VoIPPhone class requires an external SDIP server.Is this really required ?
Does anybody have an expierence with a similar project ?
I first wanted to use the linphone library, but it does not exist for Raspberry Pi.

Thanks for a all help
Marc

**Comment by tayler6000** [comment]
pyVoIP typically requires registration to a PBX server, but it is possible to start it without registering.  However, I have found that Grandstream products want to register with a PBX and do not allow contacts to be direct IP typically.

A free and easy-to-use PBX is Asterisk.  You can easily install it on Raspberry Pi and it works great with pyVoIP and Grandstream.
